### ISSUE TITLE
About batch_size 

### ISSUE THREAD

**Issue by duongkstn** [SKRohit/Generating_Text_Summary_With_GPT2#8]
Hi, your [notebook](https://github.com/SKRohit/Generating_Text_Summary_With_GPT2/blob/master/train_gpt2_summarizer.ipynb) is an impressive tutorial about GPT-2 for seq2seq model.
But it can not run with `batch_size` greater than 1. I think it is due to you added `sum_idx` in training dataset. In more details, you wrote `shift_labels = labels[..., batch['sum_idx']+1:].contiguous()` but it can not be applied for a batch since PyTorch does not enable tensor slicing by another tensor like this.

I dont know why you chose `batch_size=1`, but I can run greater `batch_size` in my NVIDIA 12GB GPU. To do that, I think you should define a new feature for your dataset, say `labels` and let some values of `labels` equals to `-100`. And you should remove `sum_idx` feature since it is really hard to slice a tensor. By assigning some values as `-100`, PyTorch's cross entroy loss will ignore those indices. Check out [`ignore_index`](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html) for more details
